addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,718 @@
+            })
+            if move.partner_id.country_code != 'GR':  # counterpart not from Greece (requires name & address)
+                party_vals.update({
+                    'counterpart_name': move.partner_id.name.encode('utf-8'),
+                    'counterpart_postal_code': move.partner_id.zip,
+                    'counterpart_city': move.partner_id.city.encode('utf-8'),
+                })
+            if move.l10n_gr_edi_inv_type in TYPES_WITH_MANDATORY_COUNTERPART:  # may be from Greece but require address
+                party_vals.update({
+                    'counterpart_postal_code': move.partner_id.zip,
+                    'counterpart_city': move.partner_id.city.encode('utf-8'),
+                })
+
+        return party_vals
+
+    @api.model
+    def _l10n_gr_edi_get_payment_method_vals(self, move):
+        """
+        Collects payment values needed for the ``invoice_vals`` dictionary.
+        Returns dictionary in the format of: { 'payment_details': [ { 'type': <str>, 'amount': <float> }, ... ] }
+        :param account.move move:
+        :rtype: dict[str, list[dict]]
+        """
+        payment_vals = {'payment_details': []}
+        reconciled_lines = move.line_ids.filtered(
+            lambda line: line.account_id.account_type in ('asset_receivable', 'liability_payable'))
+
+        for apr in reconciled_lines.matched_credit_ids:  # for out_invoice/refund payment
+            payment_vals['payment_details'].append({
+                'type': move.l10n_gr_edi_payment_method or '1',
+                'amount': apr.debit_amount_currency,
+            })
+        for apr in reconciled_lines.matched_debit_ids:  # for in_invoice/refund payment
+            payment_vals['payment_details'].append({
+                'type': move.l10n_gr_edi_payment_method or '1',
+                'amount': apr.credit_amount_currency,
+            })
+        if not payment_vals['payment_details'] and move.l10n_gr_edi_inv_type in TYPES_WITH_MANDATORY_PAYMENT:
+            # paymentMethods element is required, even if its amount is zero (no payment have been made yet)
+            payment_vals['payment_details'].append({
+                'type': move.l10n_gr_edi_payment_method or '1',
+                'amount': 0,
+            })
+
+        return payment_vals
+
+    @api.model
+    def _l10n_gr_edi_get_val_category_vals(self, line):
+        """
+        Returns additional VAT classification items needed for the detail values.
+        The returned format is: {'vat_category': <int>, 'vat_exemption_category': <str>}
+        :param account.move.line line:
+        :rtype: dict[str, int|str]
+        """
+        vat_vals = {'vat_category': 8, 'vat_exemption_category': ''}
+
+        if line.tax_ids and line.move_id.l10n_gr_edi_inv_type not in TYPES_WITH_VAT_EXEMPT:
+            vat_vals['vat_category'] = {24: 1, 13: 2, 6: 3, 17: 4, 9: 5, 4: 6, 0: 7}[int(line.tax_ids.amount)]
```

Comment:
You can have multiple taxes and also group of taxes. If the amount in not on your list, it will raise too